src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallAppsTask.groovy

```diff
@@ -212,13 +226,76 @@ class InstallAppsTask extends AbstractServerTask {
-      Pattern pattern = Pattern.compile("/build/.*")
-      Matcher matcher = pattern.matcher(dependencyPathPortion)
-      projectPath = matcher.replaceAll("")
-      return projectPath;
+    protected void installLooseConfigEar(LooseConfigData config, Task task) throws Exception{
+        LooseEarApplication looseEar = new LooseEarApplication(task, config);
+        looseEar.addSourceDir();
+        looseEar.addApplicationXmlFile();
+
+        File[] filesAsDeps = task.getProject().configurations.deploy.getFiles().toArray()
+        Dependency[] deps = task.getProject().configurations.deploy.getAllDependencies().toArray()
+        HashMap<File, Dependency> completeDeps = new HashMap<File, Dependency>();
+        if(filesAsDeps.size() == deps.size()){
+            for(int i = 0; i<filesAsDeps.size(); i++) {
+                completeDeps.put(filesAsDeps[i], deps[i])
+            }
+        }
+
+        logger.info(MessageFormat.format("Number of compile dependencies for " + task.project.name + " : " + completeDeps.size()))
+        for (Map.Entry<File, Dependency> entry : completeDeps){
+            Dependency dependency = entry.getValue();
+            File dependancyFile = entry.getKey();
+            
+            if (dependency instanceof ProjectDependency) {
+                Project dependencyProject = dependency.getDependencyProject()
+                String projectType = FilenameUtils.getExtension(dependancyFile.toString())
+                switch (projectType) {
+                        case "jar":
+                        case "ejb":
+                            looseEar.addJarModule(dependencyProject)
+                            break;
+                        case "war":
+                            Element warElement = looseEar.addWarModule(dependencyProject)
+                            addEmbeddedLib(warElement, dependencyProject, looseEar, "/WEB-INF/lib/")
+                            break;
+                        default:
+                            logger.warn('Application ' + dependencyProject.getName() + ' is expressed as ' + projectType + ' which is not a supported input type. Define applications using Task or File objects of type war, ear, or jar.')
+                            break;
+                    }
+            }
+            else if (dependency instanceof ExternalModuleDependency) {
+                String dependencyName = dependency.getName() + "-" + dependency.getVersion()
+                task.getProject().configurations.deploy.getFiles().each {
+                    if( dependencyName.equals(FilenameUtils.removeExtension(it.getName()))) {
+                        looseEar.getConfig().addFile(it.getAbsolutePath(), "/WEB-INF/lib/" + it.getName())
+                    }
+                }
+            }
+            else {
+                logger.warn("Dependency " + dependency.getName() + "could not be added to the looseApplication, as it is neither a ProjectDependency or ExternalModuleDependency")
```

Comment:
The application will not function correctly if we get here right? May need stronger message.